server/v2/stf/stf.go

```diff
@@ -401,11 +410,13 @@ func (s STF[T]) validatorUpdates(
 	return ctx.events, valSetUpdates, nil
 }
 
-const headerInfoPrefix = 0x0
+const headerInfoPrefix = 0x37
 
 // setHeaderInfo sets the header info in the state to be used by queries in the future.
 func (s STF[T]) setHeaderInfo(state store.WriterMap, headerInfo header.Info) error {
-	runtimeStore, err := state.GetWriter(appmanager.RuntimeIdentity)
+	// TODO storing header info is too low level here, stf should be stateless.
+	// We should have a keeper that does this.
+	runtimeStore, err := state.GetWriter(Identity)
```

Comment:
From local testing, this will break as-is with
```
failed to start servers: error during handshake: error on replay: unable to commit the changeset: failed to write batch to SC store: store key stf not found in multiTrees
```

consider cherry picking https://github.com/cosmos/cosmos-sdk/pull/20412/commits/fb32eafdf1761929254664f10041c0333c85fe8c to fix